src/openrct2/windows/ride.c

```diff
@@ -1834,7 +1834,9 @@ static void window_ride_init_viewport(rct_window *w)
 		rct_ride_entry* ride_entry = get_ride_entry_by_ride(ride);
 		if (ride_entry && ride_entry->tab_vehicle != 0){
 			rct_vehicle* vehicle = GET_VEHICLE(focus.sprite.sprite_id);
-			focus.sprite.sprite_id = vehicle->next_vehicle_on_train;
+			if (vehicle->next_vehicle_on_train < countof(ride_entry->vehicles)) {
```

Comment:
`countof(ride_entry->vehicles)` will always be 4. Shouldn't this be checking if `vehicle->next_vehicle_on_train` is a sprite which exists and is a vehicle sprite?